### PR TITLE
[Coral-Visualization] Create coral-visualization module and add simple test case

### PR DESCRIPTION
**Pull request by wmoustafa**
### What changes are proposed in this pull request, and why are they necessary?
This PR introduces a new module `coral-visualization` that visualizes Coral `SqlNode` IR as a graph representation using graph-vis, and writes the image to an output file. In the future, this module can be used to depict Coral RelNode IR as well.

### How was this patch tested?
Added a test case for simple query.

### CODE REVIEW

File: coral-visualization/build.gradle [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  compile project(path: ':coral-hive')
+  compile (deps.'graph-vis')
+  compile (deps.'hive'.'hive-exec-core') {
+    exclude group: 'org.apache.avro', module: 'avro-tools'
+    // These exclusions are required to prevent duplicate classes since we include
+    // shaded jar above
+    exclude group: 'org.apache.calcite', module: 'calcite-core'
+    exclude group: 'org.apache.calcite', module: 'calcite-avatica'
+  }
+  compile deps.'hadoop'.'hadoop-mapreduce-client-core'
```
Comment (ljfgem) [inline coral-visualization/build.gradle:11]: It can be `testCompile`?

File: coral-visualization/build.gradle [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  compile deps.'kryo'
```
Comment (ljfgem) [inline coral-visualization/build.gradle:12]: Not necessary?

File: coral-visualization/src/main/java/com/linkedin/coral/vis/CoraIlRVisualizationUtil.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import guru.nidi.graphviz.attribute.Font;
+import guru.nidi.graphviz.engine.Format;
+import guru.nidi.graphviz.engine.Graphviz;
+import guru.nidi.graphviz.model.Factory;
+import guru.nidi.graphviz.model.Graph;
+import guru.nidi.graphviz.model.Node;
+
+
+public class CoraIlRVisualizationUtil {
+  private static final String CORAL_HIVE_TEST_DIR = "coral.hive.test.dir";
```
Comment (ljfgem) [inline coral-visualization/src/main/java/com/linkedin/coral/vis/CoraIlRVisualizationUtil.java:30]: `CORAL_VISUALIZATION_TEST_DIR`?

File: coral-visualization/src/main/java/com/linkedin/coral/vis/CoraIlRVisualizationUtil.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  private HiveToRelConverter converter;
+  private File outputDirectory;
```
Comment (ljfgem) [inline coral-visualization/src/main/java/com/linkedin/coral/vis/CoraIlRVisualizationUtil.java:32]: Can be final?

File: coral-visualization/src/main/java/com/linkedin/coral/vis/CoraIlRVisualizationUtil.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+import guru.nidi.graphviz.attribute.Font;
+import guru.nidi.graphviz.engine.Format;
+import guru.nidi.graphviz.engine.Graphviz;
+import guru.nidi.graphviz.model.Factory;
+import guru.nidi.graphviz.model.Graph;
+import guru.nidi.graphviz.model.Node;
+
+
+public class CoraIlRVisualizationUtil {
```
Comment (ljfgem) [inline coral-visualization/src/main/java/com/linkedin/coral/vis/CoraIlRVisualizationUtil.java:29]: `CoralIRVisualizationUtil`?

**Review by ljfgem — COMMENTED** [review]
Thanks for this PR @wmoustafa!
1. Do we need to modify README to include this module or create a README in this module?
2. At present, the generated image for the SQL `SELECT * FROM foo, bar WHERE a = 1` is: 
test1
Is it better to be `table scan -> join -> filter -> project` (similar to Spark plan)?

**Comment by wmoustafa** [comment]
> * Do we need to modify README to include this module or create a README in this module?
Added.
> * At present, the generated image for the SQL `SELECT * FROM foo, bar WHERE a = 1` is <image>:
>   Is it better to be `table scan -> join -> filter -> project` (similar to Spark plan)?
This will be the representation once we visualize `RelNodes`. Right now it is intentional to be like the image to closely reflect `SqlNode` structure.

File: coral-visualization/build.gradle [hunk 6]
```diff
@@ -0,0 +1,13 @@
+dependencies {
+  compile project(path: ':coral-hive')
+  compile (deps.'graph-vis')
+  compile (deps.'hive'.'hive-exec-core') {
+    exclude group: 'org.apache.avro', module: 'avro-tools'
+    // These exclusions are required to prevent duplicate classes since we include
+    // shaded jar above
```
[the remark this answers is not in the document]
Comment (wmoustafa) [reply coral-visualization/build.gradle]: Updated.

File: coral-visualization/build.gradle (hunk 1 above)
Comment (wmoustafa) [reply coral-visualization/build.gradle]: This one was dangling where initially it was part of the main code. Now removed further APIs from `main` to `test` and marked corresponding `testCompile`.

File: coral-visualization/build.gradle (hunk 2 above)
Comment (wmoustafa) [reply coral-visualization/build.gradle]: `hadoop` and `kryo` are necessary.

File: coral-visualization/src/main/java/com/linkedin/coral/vis/CoraIlRVisualizationUtil.java (hunk 3 above)
Comment (wmoustafa) [reply coral-visualization/src/main/java/com/linkedin/coral/vis/CoraIlRVisualizationUtil.java]: Updated.

File: coral-visualization/src/main/java/com/linkedin/coral/vis/CoraIlRVisualizationUtil.java (hunk 4 above)
Comment (wmoustafa) [reply coral-visualization/src/main/java/com/linkedin/coral/vis/CoraIlRVisualizationUtil.java]: Updated.

File: coral-visualization/src/main/java/com/linkedin/coral/vis/CoraIlRVisualizationUtil.java (hunk 5 above)
Comment (wmoustafa) [reply coral-visualization/src/main/java/com/linkedin/coral/vis/CoraIlRVisualizationUtil.java]: Removed Coral from the name of both classes (also simplified this class to operate on `SqlNode` directly).

**Review by ljfgem — APPROVED** [review]
LGTM, thanks @wmoustafa!

**Pull request merged**